prime-router/src/main/kotlin/azure/Event.kt

```diff
@@ -96,80 +98,52 @@ abstract class Event(val eventAction: EventAction, val at: OffsetDateTime?) {
-                    val routeTo = if (parts[5].isNotEmpty()) {
-                        parts[5].split(ROUTE_TO_SEPARATOR)
-                    } else {
-                        emptyList()
-                    }
-
-                    // convert incoming defaults serialized string to Map<String,String>
-                    val defaults = if (parts[4].isNotEmpty()) {
-                        parts[4].split(',').associate { pair ->
-                            val defaultParts = pair.split(DEFAULT_SEPARATOR)
-                            Pair(defaultParts[0], defaultParts[1])
-                        }
-                    } else {
-                        emptyMap()
-                    }
-
-                    ProcessEvent(action, reportId, options, defaults, routeTo, after)
-                }
-                else -> error("Internal Error: invalid event type: $event")
+            val message = ObjectMapper().readValue<QueueMessage>(event)
+            val at = if (!message.at.isNullOrEmpty()) {
+                OffsetDateTime.parse(message.at)
+            } else {
+                null
             }
-        }
-
-        /**
-         * Receives incoming [event] string from the queue, breaks it into segments, verifies the correct number
-         * of segments are present based on event type, and returns the list of parts.
-         */
-        private fun parseAndValidateEvent(event: String): List<String> {
-            val parts = event.split(messageDelimiter)
-            // verify the action has at least event type, action, and report id
-            if (parts.size < 3) error("Internal Error: Queue events require eventType, action, and reportId.")
-            when (parts[0]) {
-                // Report event requires 'event type', 'action', and 'report id'. 'at' is optional
+            return when (message.eventType) {
                 ReportEvent.eventType -> {
-                    if (parts.size > 5) error("Internal Error: Report events can have no more than 4 parts.")
+                    if (message.reportId == null || message.emptyBatch == null) {
+                        error("Internal Error: Report events require a report id and is Empty Batch")
+                    }
+                    ReportEvent(
+                        message.eventAction,
+                        message.reportId,
+                        message.emptyBatch,
+                        at
+                    )
                 }
-                // Receiver event requires 'event type', 'action', 'receiver name'. 'at' is optional
                 BatchEvent.eventType -> {
-                    if (parts.size > 5) error("Internal Error: Batch events can have no more than 5 parts.")
+                    if (message.emptyBatch == null || message.receiverName.isNullOrEmpty()) {
+                        error("Internal Error: Batch events require a receiver name and is empty batch")
```

Comment:
Maybe append the "event" param at the end of this msg (and the other ones) for more helpful debugging?